#### Merge LA_Call Data with Neighborhood Council geometry

In [1]:
import pandas as pd
import numpy as np
import datetime as dt
import os
import geopandas as gpd

In [2]:
pd.options.display.max_columns = None

In [3]:
path = r'C:\Documents\projects\HackLA\311\data'

In [4]:
dfnc = pd.read_csv(os.path.join (path, 'geographic files', 'Neighborhood_Councils.csv'), index_col =  None)

In [5]:
dfnc.head()

OBJECTID                           NAME                        WADDRESS  \
0         1                      ARLETA NC        http://www.arletanc.org/   
1         2                 ARROYO SECO NC             http://www.asnc.us/   
2         3  ARTS DISTRICT LITTLE TOKYO NC          http://www.hcncla.org/   
3         4             ATWATER VILLAGE NC  http://www.atwatervillage.org/   
4         5       BEL AIR-BEVERLY CREST NC              http://babcnc.org/   

                      DWEBSITE                DEMAIL        DPHONE  NC_ID  \
0     http://empowerla.org/ANC     ANC@EmpowerLA.org  213-978-1551      6   
1    http://empowerla.org/ASNC    ASNC@EmpowerLA.org  213-978-1551     42   
2    http://empowerla.org/HCNC    HCNC@EmpowerLA.org  213-978-1551     46   
3    http://empowerla.org/AVNC    AVNC@EmpowerLA.org  213-978-1551     37   
4  http://empowerla.org/BABCNC  BABCNC@EmpowerLA.org  213-978-1551     64   

                CERTIFIED                        TOOLTIP  \
0  2002/10/22 00:00:00+00                      ARLETA NC   
1  2002/10/02 00:00:00+00                 ARROYO SECO NC   
2  2002/04/27 00:00:00+00  ARTS DISTRICT LITTLE TOKYO NC   
3  2003/02/11 00:00:00+00             ATWATER VILLAGE NC   
4  2002/10/08 00:00:00+00       BEL AIR-BEVERLY CREST NC   

                                   NLA_URL                    SERVICE_RE  
0   navigatela/reports/nc_reports.cfm?id=6  REGION 1 - NORTH EAST VALLEY  
1  navigatela/reports/nc_reports.cfm?id=42      REGION 8 - NORTH EAST LA  
2  navigatela/reports/nc_reports.cfm?id=46          REGION 6 - CENTRAL 2  
3  navigatela/reports/nc_reports.cfm?id=37               REGION 7 - EAST  
4  navigatela/reports/nc_reports.cfm?id=64           REGION 11 - WEST LA

In [6]:
dfnc.NAME.value_counts()

CENTRAL ALAMEDA NC               1
GREATER VALLEY GLEN              1
NORTHRIDGE WEST NC               1
PANORAMA CITY NC                 1
NORTH HILLS EAST NC              1
                                ..
WESTSIDE NC                      1
ARTS DISTRICT LITTLE TOKYO NC    1
DOWNTOWN LOS ANGELES             1
GREATER WILSHIRE NC              1
NORTHRIDGE SOUTH NC              1
Name: NAME, Length: 99, dtype: int64

dfnc.NAME column is the list of the 99 neighborhood councils

In [7]:
dfnc.NC_ID.count()

99

dfnc.NC_ID is the 99 neighborhood councils IDs

In [8]:
dfnc.SERVICE_RE.value_counts()

REGION 11 - WEST LA             11
REGION 6 - CENTRAL 2             9
REGION 8 - NORTH EAST LA         9
REGION 4 - SOUTH EAST VALLEY     9
REGION 1 - NORTH EAST VALLEY     9
REGION 5 - CENTRAL 1             8
REGION 3 - SOUTH WEST VALLEY     8
REGION 2 - NORTH WEST VALLEY     8
REGION 9 - SOUTH LA 2            8
REGION 12 - HARBOR               7
REGION 10 - SOUTH LA 1           7
REGION 7 - EAST                  6
Name: SERVICE_RE, dtype: int64

#### Import cleaned and wrangled 2019 crime data

In [9]:
dfcrime2019R = pd.read_csv(os.path.join (path, 'WorkedOnData', 'crime_data_2019R.csv'), index_col =  0)

In [10]:
dfcrime2019R.head()

DR_NO CreatedDate    DATE OCC  TIME OCC PolicePrecinct  \
85876  190705631  2019-02-04  2010-07-01      1100       WILSHIRE   
86769  191300738  2019-06-11  2010-06-07      1800         NEWTON   
87398  191226689  2019-11-11  2010-10-13      1500    77TH STREET   
87436  190515351  2019-09-12  2010-01-01       800         HARBOR   
87649  190907999  2019-03-14  2010-01-01         1       VAN NUYS   

       Rpt Dist No  Vict Age Vict Sex Vict Descent  \
85876          773        60        F            B   
86769         1383        23        M            B   
87398         1242        18        F            B   
87436          525        50        M            H   
87649          926        34        F            A   

                                        Premis Desc  Weapon Used Cd  \
85876                        SINGLE FAMILY DWELLING             NaN   
86769  MULTI-UNIT DWELLING (APARTMENT, DUPLEX, ETC)             NaN   
87398                        SINGLE FAMILY DWELLING             NaN   
87436                        SINGLE FAMILY DWELLING             NaN   
87649  MULTI-UNIT DWELLING (APARTMENT, DUPLEX, ETC)           400.0   

                                          Weapon Desc  Status Desc  \
85876                                             NaN  Invest Cont   
86769                                             NaN  Invest Cont   
87398                                             NaN  Invest Cont   
87436                                             NaN  Invest Cont   
87649  STRONG-ARM (HANDS, FIST, FEET OR BODILY FORCE)  Invest Cont   

                                       LOCATION  
85876   1900 S  RIDGELEY                     DR  
86769                          5800 S  BROADWAY  
87398   6500    3RD                          AV  
87436                           900    MCDONALD  
87649  14400    VICTORY                      BL

In [11]:
dfcrime2019R[dfcrime2019R.duplicated()]

Empty DataFrame
Columns: [DR_NO, CreatedDate, DATE OCC, TIME OCC, PolicePrecinct, Rpt Dist No, Vict Age, Vict Sex, Vict Descent, Premis Desc, Weapon Used Cd, Weapon Desc, Status Desc, LOCATION]
Index: []

No full row duplicates

In [12]:
dfcrime2019R.PolicePrecinct.value_counts()

77TH STREET    14312
SOUTHWEST      13705
CENTRAL        13621
PACIFIC        12345
SOUTHEAST      11877
HOLLYWOOD      11288
N HOLLYWOOD    11234
NEWTON         10814
WILSHIRE       10567
OLYMPIC        10385
TOPANGA         9757
NORTHEAST       9747
VAN NUYS        9676
RAMPART         9635
WEST LA         9348
MISSION         9285
DEVONSHIRE      9101
HARBOR          8823
WEST VALLEY     8806
HOLLENBECK      8474
FOOTHILL        7756
Name: PolicePrecinct, dtype: int64

In [13]:
dfcrimeoriginal = pd.read_csv(os.path.join (path, 'crime data', 'Crime_Data_from_2010_to_2019.csv'), index_col =  0)

In [14]:
dfcrimeoriginal.head()

Date Rptd                DATE OCC  TIME OCC  AREA   \
DR_NO                                                                        
1307355    02/20/2010 12:00:00 AM  02/20/2010 12:00:00 AM      1350     13   
11401303   09/13/2010 12:00:00 AM  09/12/2010 12:00:00 AM        45     14   
70309629   08/09/2010 12:00:00 AM  08/09/2010 12:00:00 AM      1515     13   
90631215   01/05/2010 12:00:00 AM  01/05/2010 12:00:00 AM       150      6   
100100501  01/03/2010 12:00:00 AM  01/02/2010 12:00:00 AM      2100      1   

           AREA NAME  Rpt Dist No  Part 1-2  Crm Cd  \
DR_NO                                                 
1307355       Newton         1385         2     900   
11401303     Pacific         1485         2     740   
70309629      Newton         1324         2     946   
90631215   Hollywood          646         2     900   
100100501    Central          176         1     122   

                                                 Crm Cd Desc         Mocodes  \
DR_NO                                                                          
1307355                             VIOLATION OF COURT ORDER  0913 1814 2000   
11401303   VANDALISM - FELONY ($400 & OVER, ALL CHURCH VA...            0329   
70309629                           OTHER MISCELLANEOUS CRIME            0344   
90631215                            VIOLATION OF COURT ORDER  1100 0400 1402   
100100501                                    RAPE, ATTEMPTED            0400   

           Vict Age Vict Sex Vict Descent  Premis Cd             Premis Desc  \
DR_NO                                                                          
1307355          48        M            H      501.0  SINGLE FAMILY DWELLING   
11401303          0        M            W      101.0                  STREET   
70309629          0        M            H      103.0                   ALLEY   
90631215         47        F            W      101.0                  STREET   
100100501        47        F            H      103.0                   ALLEY   

           Weapon Used Cd                                     Weapon Desc  \
DR_NO                                                                       
1307355               NaN                                             NaN   
11401303              NaN                                             NaN   
70309629              NaN                                             NaN   
90631215            102.0                                        HAND GUN   
100100501           400.0  STRONG-ARM (HANDS, FIST, FEET OR BODILY FORCE)   

          Status   Status Desc  Crm Cd 1  Crm Cd 2  Crm Cd 3  Crm Cd 4  \
DR_NO                                                                    
1307355       AA  Adult Arrest     900.0       NaN       NaN       NaN   
11401303      IC   Invest Cont     740.0       NaN       NaN       NaN   
70309629      IC   Invest Cont     946.0       NaN       NaN       NaN   
90631215      IC   Invest Cont     900.0     998.0       NaN       NaN   
100100501     IC   Invest Cont     122.0       NaN       NaN       NaN   

                                          LOCATION  \
DR_NO                                                
1307355     300 E  GAGE                         AV   
11401303           SEPULVEDA                    BL   
70309629   1300 E  21ST                         ST   
90631215           CAHUENGA                     BL   
100100501          8TH                          ST   

                              Cross Street      LAT       LON  
DR_NO                                                          
1307355                                NaN  33.9825 -118.2695  
11401303   MANCHESTER                   AV  33.9599 -118.3962  
70309629                               NaN  34.0224 -118.2524  
90631215   HOLLYWOOD                    BL  34.1016 -118.3295  
100100501  SAN PEDRO                    ST  34.0387 -118.2488

In [19]:
dfnc = gpd.read_file(r'C:\Documents\projects\HackLA\311\data\geographic files\shp\Neighborhood_Councils_(Certified).shp')

In [20]:
dfnc.head()

OBJECTID                           NAME                        WADDRESS  \
0         1                      ARLETA NC        http://www.arletanc.org/   
1         2                 ARROYO SECO NC             http://www.asnc.us/   
2         3  ARTS DISTRICT LITTLE TOKYO NC          http://www.hcncla.org/   
3         4             ATWATER VILLAGE NC  http://www.atwatervillage.org/   
4         5       BEL AIR-BEVERLY CREST NC              http://babcnc.org/   

                      DWEBSITE                DEMAIL        DPHONE  NC_ID  \
0     http://empowerla.org/ANC     ANC@EmpowerLA.org  213-978-1551      6   
1    http://empowerla.org/ASNC    ASNC@EmpowerLA.org  213-978-1551     42   
2    http://empowerla.org/HCNC    HCNC@EmpowerLA.org  213-978-1551     46   
3    http://empowerla.org/AVNC    AVNC@EmpowerLA.org  213-978-1551     37   
4  http://empowerla.org/BABCNC  BABCNC@EmpowerLA.org  213-978-1551     64   

    CERTIFIED                        TOOLTIP  \
0  2002-10-22                      ARLETA NC   
1  2002-10-02                 ARROYO SECO NC   
2  2002-04-27  ARTS DISTRICT LITTLE TOKYO NC   
3  2003-02-11             ATWATER VILLAGE NC   
4  2002-10-08       BEL AIR-BEVERLY CREST NC   

                                   NLA_URL                    SERVICE_RE  \
0   navigatela/reports/nc_reports.cfm?id=6  REGION 1 - NORTH EAST VALLEY   
1  navigatela/reports/nc_reports.cfm?id=42      REGION 8 - NORTH EAST LA   
2  navigatela/reports/nc_reports.cfm?id=46          REGION 6 - CENTRAL 2   
3  navigatela/reports/nc_reports.cfm?id=37               REGION 7 - EAST   
4  navigatela/reports/nc_reports.cfm?id=64           REGION 11 - WEST LA   

                                            geometry  
0  POLYGON ((-118.44305 34.26328, -118.44303 34.2...  
1  POLYGON ((-118.19567 34.08613, -118.19568 34.0...  
2  POLYGON ((-118.23151 34.05348, -118.23141 34.0...  
3  POLYGON ((-118.25279 34.10833, -118.25292 34.1...  
4  POLYGON ((-118.36574 34.09948, -118.36575 34.0...

We have the  neighborhood council with geometry.  The calls file has NC.  The original crime file has long and lat.  Hopefully can 
connect on long and lat from crime to NC geometry.

First let's connect the geofile with the cleaned up 2019 calls. Then redo the the crime data so it includes the long and lat and see if we can merge using that

In [23]:
df_call = pd.read_csv(os.path.join (path, 'WorkedonData', 'la_311_2019.csv'), index_col =  0) 

In [24]:
df_call.head()

SRNumber             CreatedDate             UpdatedDate     NC  \
0  1-1262692791  01/01/2019 12:02:00 AM  01/04/2019 11:03:00 AM  119.0   
1  1-1262693531  01/01/2019 12:05:00 AM  01/01/2019 12:09:00 AM    NaN   
2  1-1262693571  01/01/2019 12:10:00 AM  01/03/2019 12:27:00 AM  113.0   
3  1-1262692831  01/01/2019 12:19:00 AM  01/07/2019 09:43:00 AM  124.0   
4  1-1262692061  01/01/2019 12:28:00 AM  01/03/2019 12:28:00 AM  124.0   

                NCName             RequestType RequestSource  \
0  GREATER WILSHIRE NC             Bulky Items  Self Service   
1                  NaN      Report Water Waste    Mobile App   
2      NORTHRIDGE WEST        Graffiti Removal  Self Service   
3  NORTHRIDGE SOUTH NC  Illegal Dumping Pickup  Self Service   
4  NORTHRIDGE SOUTH NC        Graffiti Removal  Self Service   

                           Address  ZipCode   Latitude   Longitude  \
0         616 N GRAMERCY PL, 90004  90004.0  34.082258 -118.312461   
1  11500 W SAN VICENTE BLVD, 90049  90049.0  34.052739 -118.461185   
2        9167 N RESEDA BLVD, 91324  91324.0  34.236888 -118.536385   
3        8752 N YOLANDA AVE, 91324  91324.0  34.230222 -118.539758   
4        9055 N RESEDA BLVD, 91324  91324.0  34.235078 -118.536392   

                          Location PolicePrecinct  
0  (34.0822581437, -118.312461304)        OLYMPIC  
1   (34.052739298, -118.461184916)        WEST LA  
2   (34.2368883475, -118.53638542)     DEVONSHIRE  
3  (34.2302221251, -118.539758291)     DEVONSHIRE  
4  (34.2350781243, -118.536391683)     DEVONSHIRE

In [25]:
df_call.isnull().sum()

SRNumber             0
CreatedDate          0
UpdatedDate          0
NC                7231
NCName            7231
RequestType          0
RequestSource        0
Address              0
ZipCode            438
Latitude             0
Longitude            0
Location             0
PolicePrecinct       0
dtype: int64

In [26]:
df_call.shape

(1306771, 13)

In [27]:
7231/1305771

0.005537724455513256

NC nulls are .5% so safe to delete them


In [28]:
df_call.dropna(inplace=True)

In [29]:
df_call.isnull().sum()

SRNumber          0
CreatedDate       0
UpdatedDate       0
NC                0
NCName            0
RequestType       0
RequestSource     0
Address           0
ZipCode           0
Latitude          0
Longitude         0
Location          0
PolicePrecinct    0
dtype: int64

NC nulls are gone

In [33]:
df_call['NCName'].value_counts()

BOYLE HEIGHTS NC                           36689
SYLMAR NC                                  33699
VAN NUYS NC                                31200
EMPOWERMENT CONGRESS SOUTHEAST AREA NDC    30271
WILSHIRE CENTER - KOREATOWN NC             29340
                                           ...  
Winnetka                                       2
Reseda                                         1
Granada Hills South                            1
Encino                                         1
Lake Balboa                                    1
Name: NCName, Length: 118, dtype: int64

In [34]:
df_call.shape

(1299102, 13)

df_call has 1,299,102 rows

In [35]:
dfnc.NAME.value_counts()

CENTRAL ALAMEDA NC               1
GREATER VALLEY GLEN              1
NORTHRIDGE WEST NC               1
PANORAMA CITY NC                 1
NORTH HILLS EAST NC              1
                                ..
WESTSIDE NC                      1
ARTS DISTRICT LITTLE TOKYO NC    1
DOWNTOWN LOS ANGELES             1
GREATER WILSHIRE NC              1
NORTHRIDGE SOUTH NC              1
Name: NAME, Length: 99, dtype: int64

In [36]:
dfnc.shape

(99, 12)

dfnc just has the 99 rows of each council.

When we merge we should end up with 1299102 rows

In [39]:
dfnc.rename(columns = {"NAME": "NCName"}, inplace=True)

In [40]:
dfnc.head(2)

OBJECTID          NCName                  WADDRESS  \
0         1       ARLETA NC  http://www.arletanc.org/   
1         2  ARROYO SECO NC       http://www.asnc.us/   

                    DWEBSITE              DEMAIL        DPHONE  NC_ID  \
0   http://empowerla.org/ANC   ANC@EmpowerLA.org  213-978-1551      6   
1  http://empowerla.org/ASNC  ASNC@EmpowerLA.org  213-978-1551     42   

    CERTIFIED         TOOLTIP                                  NLA_URL  \
0  2002-10-22       ARLETA NC   navigatela/reports/nc_reports.cfm?id=6   
1  2002-10-02  ARROYO SECO NC  navigatela/reports/nc_reports.cfm?id=42   

                     SERVICE_RE  \
0  REGION 1 - NORTH EAST VALLEY   
1      REGION 8 - NORTH EAST LA   

                                            geometry  
0  POLYGON ((-118.44305 34.26328, -118.44303 34.2...  
1  POLYGON ((-118.19567 34.08613, -118.19568 34.0...

In [41]:
df_call['NCName']=df_call['NCName'].str.upper()

In [42]:
df_call.head(2)

SRNumber             CreatedDate             UpdatedDate     NC  \
0  1-1262692791  01/01/2019 12:02:00 AM  01/04/2019 11:03:00 AM  119.0   
2  1-1262693571  01/01/2019 12:10:00 AM  01/03/2019 12:27:00 AM  113.0   

                NCName       RequestType RequestSource  \
0  GREATER WILSHIRE NC       Bulky Items  Self Service   
2      NORTHRIDGE WEST  Graffiti Removal  Self Service   

                     Address  ZipCode   Latitude   Longitude  \
0   616 N GRAMERCY PL, 90004  90004.0  34.082258 -118.312461   
2  9167 N RESEDA BLVD, 91324  91324.0  34.236888 -118.536385   

                          Location PolicePrecinct  
0  (34.0822581437, -118.312461304)        OLYMPIC  
2   (34.2368883475, -118.53638542)     DEVONSHIRE

In [43]:
df_call.NCName.value_counts()

BOYLE HEIGHTS NC                           36689
SYLMAR NC                                  33699
VAN NUYS NC                                31200
EMPOWERMENT CONGRESS SOUTHEAST AREA NDC    30271
WILSHIRE CENTER - KOREATOWN NC             29340
                                           ...  
NORTH HILLS WEST                               2
ENCINO                                         1
RESEDA                                         1
GRANADA HILLS SOUTH                            1
LAKE BALBOA                                    1
Name: NCName, Length: 117, dtype: int64

In [46]:
NameDiff = np.setdiff1d(df_call.NCName,dfnc.NCName)

In [47]:
NameDiff

array(['BRENTWOOD CC', 'CANOGA PARK', 'CHATSWORTH',
       'EMPOWERMENT CONGRESS CENTRAL AREA NDC',
       'EMPOWERMENT CONGRESS NORTH AREA NDC', 'ENCINO',
       'GRANADA HILLS SOUTH', 'GREATER ECHO PARK ELYSIAN NC',
       'GREATER VALLEY GLEN COUNCIL', 'HISTORIC CULTURAL NC',
       'HISTORIC FILIPINOTOWN NC', 'LAKE BALBOA', 'MAR VISTA CC',
       'MID-TOWN NORTH HOLLYWOOD NC', 'NC WESTCHESTER/PLAYA',
       'NORTH HILLS EAST', 'NORTH HILLS WEST',
       'NORTH HOLLYWOOD NORTHEAST NC', 'NORTHRIDGE EAST',
       'NORTHRIDGE WEST', 'OLD NORTHRIDGE CC', 'PACIFIC PALISADES NC',
       'PARK MESA HEIGHTS CC', 'PORTER RANCH', 'RESEDA',
       'UNITED FOR VICTORY', 'UNITED NEIGHBORHOODS',
       'UNITED NEIGHBORHOODS OF THE HISTORIC ARLINGTON HEIGHTS, WEST ADAMS, AND JEFFERSON PARK COMMUNITY',
       'VOICES OF 90037', 'WEST HILLS', 'WEST LOS ANGELES NC', 'WINNETKA',
       'ZAPATA KING NC'], dtype=object)

The NCNames values are not exactly alike.  Will look at NC codes

dfnc.NC_ID & df_call.NC

In [48]:
dfnc.NC_ID.dtype

dtype('int64')

In [49]:
df_call.NC.dtype

dtype('float64')

In [52]:
df_call['NC'] = df_call['NC'].astype(int)

In [53]:
df_call.head()

SRNumber             CreatedDate             UpdatedDate   NC  \
0  1-1262692791  01/01/2019 12:02:00 AM  01/04/2019 11:03:00 AM  119   
2  1-1262693571  01/01/2019 12:10:00 AM  01/03/2019 12:27:00 AM  113   
3  1-1262692831  01/01/2019 12:19:00 AM  01/07/2019 09:43:00 AM  124   
4  1-1262692061  01/01/2019 12:28:00 AM  01/03/2019 12:28:00 AM  124   
5  1-1262694421  01/01/2019 12:32:00 AM  01/03/2019 04:27:00 PM  120   

                NCName             RequestType RequestSource  \
0  GREATER WILSHIRE NC             Bulky Items  Self Service   
2      NORTHRIDGE WEST        Graffiti Removal  Self Service   
3  NORTHRIDGE SOUTH NC  Illegal Dumping Pickup  Self Service   
4  NORTHRIDGE SOUTH NC        Graffiti Removal  Self Service   
5      NORTHRIDGE EAST        Graffiti Removal  Self Service   

                     Address  ZipCode   Latitude   Longitude  \
0   616 N GRAMERCY PL, 90004  90004.0  34.082258 -118.312461   
2  9167 N RESEDA BLVD, 91324  91324.0  34.236888 -118.536385   
3  8752 N YOLANDA AVE, 91324  91324.0  34.230222 -118.539758   
4  9055 N RESEDA BLVD, 91324  91324.0  34.235078 -118.536392   
5   9312 N ZELZAH AVE, 91325  91325.0  34.239704 -118.523064   

                          Location PolicePrecinct  
0  (34.0822581437, -118.312461304)        OLYMPIC  
2   (34.2368883475, -118.53638542)     DEVONSHIRE  
3  (34.2302221251, -118.539758291)     DEVONSHIRE  
4  (34.2350781243, -118.536391683)     DEVONSHIRE  
5  (34.2397037866, -118.523064396)     DEVONSHIRE

In [54]:
df_call.rename(columns = {"NC": "NC_ID"}, inplace=True)

In [55]:
df_call.head(2)

SRNumber             CreatedDate             UpdatedDate  NC_ID  \
0  1-1262692791  01/01/2019 12:02:00 AM  01/04/2019 11:03:00 AM    119   
2  1-1262693571  01/01/2019 12:10:00 AM  01/03/2019 12:27:00 AM    113   

                NCName       RequestType RequestSource  \
0  GREATER WILSHIRE NC       Bulky Items  Self Service   
2      NORTHRIDGE WEST  Graffiti Removal  Self Service   

                     Address  ZipCode   Latitude   Longitude  \
0   616 N GRAMERCY PL, 90004  90004.0  34.082258 -118.312461   
2  9167 N RESEDA BLVD, 91324  91324.0  34.236888 -118.536385   

                          Location PolicePrecinct  
0  (34.0822581437, -118.312461304)        OLYMPIC  
2   (34.2368883475, -118.53638542)     DEVONSHIRE

In [56]:
NCDiff = np.setdiff1d(df_call.NC_ID,dfnc.NC_ID)

In [57]:
NCDiff

array([  0,   2,  65, 117, 122, 123])

In [59]:
np.sort(dfnc.NC_ID.unique())

array([  4,   5,   6,   7,   8,   9,  10,  11,  13,  14,  15,  16,  17,
        18,  19,  20,  21,  22,  23,  24,  25,  26,  27,  28,  29,  30,
        32,  33,  34,  36,  37,  38,  39,  40,  41,  42,  43,  44,  46,
        47,  48,  50,  52,  53,  54,  55,  58,  60,  61,  62,  63,  64,
        66,  67,  68,  70,  71,  73,  74,  75,  76,  77,  78,  79,  80,
        81,  84,  86,  87,  88,  90,  91,  92,  93,  94,  95,  96,  97,
        99, 100, 101, 102, 104, 109, 110, 111, 112, 113, 114, 115, 118,
       119, 120, 121, 124, 125, 126, 127, 128], dtype=int64)

In [60]:
np.sort(df_call.NC_ID.unique())

array([  0,   2,   4,   5,   6,   7,   8,   9,  10,  11,  13,  14,  15,
        16,  17,  18,  19,  20,  21,  22,  23,  24,  25,  26,  27,  28,
        29,  30,  32,  33,  34,  36,  37,  38,  39,  40,  41,  42,  43,
        44,  46,  47,  48,  50,  52,  53,  54,  55,  58,  60,  61,  62,
        63,  64,  65,  66,  67,  68,  70,  71,  73,  74,  75,  76,  77,
        78,  79,  80,  81,  84,  86,  87,  88,  90,  91,  92,  93,  94,
        95,  96,  97,  99, 100, 101, 102, 104, 109, 110, 111, 112, 113,
       114, 115, 117, 118, 119, 120, 121, 122, 123, 124, 125, 126, 127,
       128])

The ID numbers in the neiborhood council ID list is correct (says "verified" on the website) so will hunt down the incorrect numbers in the call dataset and see if can correct them.  If not, see if they're a small enough percentage to delete


In [67]:
#This is a df made up of all IDs == 0
df0 = df_call[df_call.NC_ID == 0]

In [68]:
df0.NCName.value_counts()

HISTORIC CULTURAL NORTH NC    4610
NORTH WESTWOOD NC             1849
Name: NCName, dtype: int64

historic cultural north nc should be 128

In [74]:
df_call[df_call.NCName.isin(['HISTORIC CULTURAL NORTH NC'])]

SRNumber             CreatedDate             UpdatedDate  NC_ID  \
402182   1-1345104701  05/03/2019 10:09:00 AM  05/03/2019 11:23:00 AM      0   
402186   1-1345110051  05/03/2019 10:09:00 AM  05/07/2019 05:10:00 PM      0   
402209   1-1345113461  05/03/2019 10:15:00 AM  05/07/2019 05:08:00 PM      0   
403364   1-1345358120  05/03/2019 01:45:00 PM  05/08/2019 06:37:00 AM      0   
403726   1-1345464211  05/03/2019 03:17:00 PM  05/08/2019 06:38:00 AM      0   
...               ...                     ...                     ...    ...   
1307580  1-1523514809  12/31/2019 04:03:00 PM  12/31/2019 04:05:00 PM    128   
1307604  1-1523516051  12/31/2019 04:07:00 PM  12/31/2019 04:10:00 PM    128   
1307610  1-1523516069  12/31/2019 04:07:00 PM  12/31/2019 04:10:00 PM    128   
1307618  1-1523522891  12/31/2019 04:09:00 PM  01/02/2020 07:46:00 AM    128   
1307958  1-1523564401  12/31/2019 07:31:00 PM  01/08/2020 04:03:00 PM    128   

                             NCName             RequestType  \
402182   HISTORIC CULTURAL NORTH NC     Dead Animal Removal   
402186   HISTORIC CULTURAL NORTH NC        Graffiti Removal   
402209   HISTORIC CULTURAL NORTH NC        Graffiti Removal   
403364   HISTORIC CULTURAL NORTH NC             Bulky Items   
403726   HISTORIC CULTURAL NORTH NC  Illegal Dumping Pickup   
...                             ...                     ...   
1307580  HISTORIC CULTURAL NORTH NC        Graffiti Removal   
1307604  HISTORIC CULTURAL NORTH NC        Graffiti Removal   
1307610  HISTORIC CULTURAL NORTH NC        Graffiti Removal   
1307618  HISTORIC CULTURAL NORTH NC             Bulky Items   
1307958  HISTORIC CULTURAL NORTH NC        Graffiti Removal   

              RequestSource                          Address  ZipCode  \
402182                 Call       1263 N BISHOPS ROAD, 90012  90012.0   
402186                 Call  819 E CESAR E CHAVEZ AVE, 90012  90012.0   
402209                 Call  819 W CESAR E CHAVEZ AVE, 90012  90012.0   
403364           Mobile App             700 N HILL PL, 90012  90012.0   
403726         Self Service          714 W LOOKOUT DR, 90012  90012.0   
...                     ...                              ...      ...   
1307580  Driver Self Report     759 N NORTH SPRING ST, 90012  90012.0   
1307604  Driver Self Report     715 N NORTH SPRING ST, 90012  90012.0   
1307610  Driver Self Report     821 N NORTH SPRING ST, 90012  90012.0   
1307618                Call        995 N FIGUEROA TER, 90012  90012.0   
1307958          Mobile App  BISHOPS ROAD AT SAVOY ST, 90012  90012.0   

          Latitude   Longitude                         Location PolicePrecinct  
402182   34.070310 -118.234847  (34.0703103014, -118.234847409)        CENTRAL  
402186   34.055429 -118.228162  (34.0554291992, -118.228161886)        CENTRAL  
402209   34.061692 -118.244939  (34.0616920407, -118.244938751)        CENTRAL  
403364   34.061615 -118.242320  (34.0616146648, -118.242319892)        CENTRAL  
403726   34.068295 -118.240278  (34.0682946292, -118.240278298)        CENTRAL  
...            ...         ...                              ...            ...  
1307580  34.061448 -118.237311  (34.0614479462, -118.237310645)        CENTRAL  
1307604  34.060291 -118.237871  (34.0602905815, -118.237871161)        CENTRAL  
1307610  34.062166 -118.236976  (34.0621658341, -118.236975675)        CENTRAL  
1307618  34.068410 -118.244830  (34.0684101694, -118.244829876)        CENTRAL  
1307958  34.069595 -118.233979   (34.069595341, -118.233978989)        CENTRAL  

[5551 rows x 13 columns]

In [86]:
#run test replace
df0.loc[df0['NCName'] == 'HISTORIC CULTURAL NORTH NC', 'NC_ID'] = 128


c:\ProgramData\Anaconda3\envs\geo_env\lib\site-packages\pandas\core\indexing.py:1720: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  self._setitem_single_column(loc, value, pi)


In [87]:
df0.NC_ID.value_counts()

128    4610
0      1849
Name: NC_ID, dtype: int64

In [88]:
df0.head()

SRNumber             CreatedDate             UpdatedDate NC_ID  \
402182  1-1345104701  05/03/2019 10:09:00 AM  05/03/2019 11:23:00 AM   128   
402186  1-1345110051  05/03/2019 10:09:00 AM  05/07/2019 05:10:00 PM   128   
402209  1-1345113461  05/03/2019 10:15:00 AM  05/07/2019 05:08:00 PM   128   
402573  1-1345192821  05/03/2019 11:14:00 AM  05/08/2019 09:56:00 AM     0   
402584  1-1345194321  05/03/2019 11:16:00 AM  05/08/2019 06:39:00 AM     0   

                            NCName                 RequestType RequestSource  \
402182  HISTORIC CULTURAL NORTH NC         Dead Animal Removal          Call   
402186  HISTORIC CULTURAL NORTH NC            Graffiti Removal          Call   
402209  HISTORIC CULTURAL NORTH NC            Graffiti Removal          Call   
402573           NORTH WESTWOOD NC  Metal/Household Appliances          Call   
402584           NORTH WESTWOOD NC                 Bulky Items          Call   

                                Address  ZipCode   Latitude   Longitude  \
402182       1263 N BISHOPS ROAD, 90012  90012.0  34.070310 -118.234847   
402186  819 E CESAR E CHAVEZ AVE, 90012  90012.0  34.055429 -118.228162   
402209  819 W CESAR E CHAVEZ AVE, 90012  90012.0  34.061692 -118.244939   
402573        619 S LANDFAIR AVE, 90024  90024.0  34.066715 -118.449241   
402584        619 S LANDFAIR AVE, 90024  90024.0  34.066715 -118.449241   

                               Location PolicePrecinct  
402182  (34.0703103014, -118.234847409)        CENTRAL  
402186  (34.0554291992, -118.228161886)        CENTRAL  
402209  (34.0616920407, -118.244938751)        CENTRAL  
402573  (34.0667149459, -118.449241006)        WEST LA  
402584  (34.0667149459, -118.449241006)        WEST LA

Replacement of NC_ID 0 with 128 for HISTORIC CULTURAL NORTH NC in test df worked

NORTH WESTWOOD NC should be 127

In [89]:
df0.loc[df0['NCName'] == 'NORTH WESTWOOD NC', 'NC_ID'] = 127

c:\ProgramData\Anaconda3\envs\geo_env\lib\site-packages\pandas\core\indexing.py:1720: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  self._setitem_single_column(loc, value, pi)


In [91]:
df0.NC_ID.value_counts()

128    4610
127    1849
Name: NC_ID, dtype: int64

CODE worked on test df.  Will run it on df_call

In [92]:
df_call.loc[df_call['NCName'] == 'NORTH WESTWOOD NC', 'NC_ID'] = 127

In [93]:
df_call.loc[df_call['NCName'] == 'HISTORIC CULTURAL NORTH NC', 'NC_ID'] = 128

In [95]:
df_call[df_call.NC_ID == 0]

Empty DataFrame
Columns: [SRNumber, CreatedDate, UpdatedDate, NC_ID, NCName, RequestType, RequestSource, Address, ZipCode, Latitude, Longitude, Location, PolicePrecinct]
Index: []

In [96]:
NCDiff

array([  0,   2,  65, 117, 122, 123])

The above is the set of original differences.  Let's look at '2' now

In [97]:
df_call[df_call.NC_ID == 2]

SRNumber             CreatedDate             UpdatedDate  NC_ID  \
461282   1-1357208731  05/21/2019 10:54:00 AM  05/22/2019 03:01:00 PM      2   
511312   1-1367339431  06/04/2019 03:09:00 PM  06/06/2019 01:04:00 AM      2   
511409   1-1367361991  06/04/2019 03:24:00 PM  06/06/2019 01:11:00 AM      2   
613413   1-1387843831  07/01/2019 12:32:00 PM  07/03/2019 01:39:00 AM      2   
819950   1-1429493361  08/21/2019 10:44:00 AM  08/27/2019 02:38:00 PM      2   
861926   1-1437927811  09/02/2019 10:28:00 AM  09/03/2019 03:37:00 PM      2   
916093   1-1448858001  09/15/2019 02:10:00 PM  09/16/2019 07:59:00 PM      2   
1049108  1-1473397841  10/19/2019 01:48:00 PM  10/21/2019 10:39:00 AM      2   
1197623  1-1502201591  11/28/2019 09:29:00 PM  12/02/2019 07:07:00 AM      2   
1202338  1-1503340531  12/01/2019 02:09:00 PM  12/02/2019 12:10:00 PM      2   

                    NCName          RequestType RequestSource  \
461282   OLD NORTHRIDGE CC     Graffiti Removal          Call   
511312   OLD NORTHRIDGE CC     Graffiti Removal          Call   
511409   OLD NORTHRIDGE CC     Graffiti Removal          Call   
613413   OLD NORTHRIDGE CC     Graffiti Removal          Call   
819950   OLD NORTHRIDGE CC  Homeless Encampment  Self Service   
861926   OLD NORTHRIDGE CC     Graffiti Removal          Call   
916093   OLD NORTHRIDGE CC     Graffiti Removal          Call   
1049108  OLD NORTHRIDGE CC     Graffiti Removal          Call   
1197623  OLD NORTHRIDGE CC  Homeless Encampment          Call   
1202338  OLD NORTHRIDGE CC     Graffiti Removal  Self Service   

                                 Address  ZipCode   Latitude   Longitude  \
461282   TAMPA AVE AT ROSCOE BLVD, 91324  91324.0  34.220453 -118.553596   
511312   ROSCOE BLVD AT TAMPA AVE, 91324  91324.0  34.220453 -118.553596   
511409   ROSCOE BLVD AT TAMPA AVE, 91324  91324.0  34.220453 -118.553596   
613413   TAMPA AVE AT ROSCOE BLVD, 91324  91324.0  34.220453 -118.553596   
819950   ROSCOE BLVD AT TAMPA AVE, 91324  91324.0  34.220453 -118.553596   
861926   TAMPA AVE AT ROSCOE BLVD, 91324  91324.0  34.220453 -118.553596   
916093   TAMPA AVE AT ROSCOE BLVD, 91324  91324.0  34.220453 -118.553596   
1049108  ROSCOE BLVD AT TAMPA AVE, 91324  91324.0  34.220453 -118.553596   
1197623  ROSCOE BLVD AT TAMPA AVE, 91324  91324.0  34.220453 -118.553596   
1202338  TAMPA AVE AT ROSCOE BLVD, 91324  91324.0  34.220453 -118.553596   

                                Location PolicePrecinct  
461282   (34.2204525568, -118.553595768)    WEST VALLEY  
511312   (34.2204525568, -118.553595768)    WEST VALLEY  
511409   (34.2204525568, -118.553595768)    WEST VALLEY  
613413   (34.2204525568, -118.553595768)    WEST VALLEY  
819950   (34.2204525568, -118.553595768)    WEST VALLEY  
861926   (34.2204525568, -118.553595768)    WEST VALLEY  
916093   (34.2204525568, -118.553595768)    WEST VALLEY  
1049108  (34.2204525568, -118.553595768)    WEST VALLEY  
1197623  (34.2204525568, -118.553595768)    WEST VALLEY  
1202338  (34.2204525568, -118.553595768)    WEST VALLEY

OLD NORTHRIDGE CC doesn't exist anymore and these rows are a tiny percentage of the df_call dataset so will delete them

In [99]:
df_call[df_call.NC_ID == 2].count()

SRNumber          10
CreatedDate       10
UpdatedDate       10
NC_ID             10
NCName            10
RequestType       10
RequestSource     10
Address           10
ZipCode           10
Latitude          10
Longitude         10
Location          10
PolicePrecinct    10
dtype: int64

In [100]:
df_call.shape

(1299102, 13)

In [106]:
df_call.drop(df_call.loc[df_call['NC_ID'] == 2].index, inplace=True)

In [108]:
df_call[df_call.NC_ID == 2]

Empty DataFrame
Columns: [SRNumber, CreatedDate, UpdatedDate, NC_ID, NCName, RequestType, RequestSource, Address, ZipCode, Latitude, Longitude, Location, PolicePrecinct]
Index: []

In [109]:
df_call.shape

(1299092, 13)

All rows with NC_ID = 2 are deleted

In [111]:
df_call[df_call.NC_ID == 65]

SRNumber             CreatedDate             UpdatedDate  NC_ID  \
394429   1-1343535711  05/01/2019 12:15:00 PM  05/07/2019 03:59:00 PM     65   
402047   1-1345073031  05/03/2019 09:38:00 AM  05/08/2019 03:44:00 PM     65   
402729   1-1345224941  05/03/2019 11:40:00 AM  05/07/2019 04:07:00 PM     65   
402734   1-1345225971  05/03/2019 11:41:00 AM  05/07/2019 01:04:00 PM     65   
402974   1-1345262891  05/03/2019 12:14:00 PM  05/08/2019 10:09:00 AM     65   
...               ...                     ...                     ...    ...   
1306656  1-1523288421  12/31/2019 12:48:00 PM  01/02/2020 08:47:00 AM     65   
1307282  1-1523454872  12/31/2019 03:11:00 PM  01/07/2020 07:21:00 AM     65   
1307500  1-1523501971  12/31/2019 03:52:00 PM  01/07/2020 04:14:00 PM     65   
1307528  1-1523511551  12/31/2019 03:58:00 PM  01/15/2020 06:58:00 PM     65   
1307749  1-1523535579  12/31/2019 04:42:00 PM  01/02/2020 01:27:00 PM     65   

               NCName                 RequestType RequestSource  \
394429   BRENTWOOD CC                 Bulky Items          Call   
402047   BRENTWOOD CC                 Bulky Items          Call   
402729   BRENTWOOD CC                 Bulky Items          Call   
402734   BRENTWOOD CC  Metal/Household Appliances          Call   
402974   BRENTWOOD CC                 Bulky Items          Call   
...               ...                         ...           ...   
1306656  BRENTWOOD CC                 Bulky Items  Self Service   
1307282  BRENTWOOD CC                 Bulky Items    Mobile App   
1307500  BRENTWOOD CC                 Bulky Items          Call   
1307528  BRENTWOOD CC            Graffiti Removal  Self Service   
1307749  BRENTWOOD CC                 Bulky Items    Mobile App   

                                          Address  ZipCode   Latitude  \
394429                  470 N ROBINWOOD DR, 90049  90049.0  34.067549   
402047                  11828 W GOSHEN AVE, 90049  90049.0  34.048313   
402729                  11630 W GORHAM AVE, 90049  90049.0  34.055018   
402734                  11630 W GORHAM AVE, 90049  90049.0  34.055018   
402974                  11938 W GOSHEN AVE, 90049  90049.0  34.046853   
...                                           ...      ...        ...   
1306656  SEPULVEDA BLVD AT GETTY CENTER DR, 90049  90049.0  34.088644   
1307282                  11350 W ALBATA ST, 90049  90049.0  34.064110   
1307500               527 S BARRINGTON AVE, 90049  90049.0  34.058872   
1307528      SEPULVEDA BLVD AT BERGREEN PL, 90049  90049.0  34.124961   
1307749                 12012 W GOSHEN AVE, 90049  90049.0  34.045986   

          Longitude                         Location PolicePrecinct  
394429  -118.482733  (34.0675489321, -118.482732632)        WEST LA  
402047  -118.463982   (34.048313367, -118.463982049)        WEST LA  
402729  -118.464567    (34.055018224, -118.46456747)        WEST LA  
402734  -118.464567    (34.055018224, -118.46456747)        WEST LA  
402974  -118.465769  (34.0468527452, -118.465768646)        WEST LA  
...             ...                              ...            ...  
1306656 -118.474848  (34.0886442909, -118.474848029)        WEST LA  
1307282 -118.460473  (34.0641097743, -118.460472788)        WEST LA  
1307500 -118.468686  (34.0588723875, -118.468685848)        WEST LA  
1307528 -118.478516  (34.1249614241, -118.478516534)        WEST LA  
1307749 -118.466846  (34.0459860386, -118.466845713)        WEST LA  

[3373 rows x 13 columns]

BRENTWOOD CC does exist anymore and it's only 13 rows.  Will delete.

In [112]:
df_call.drop(df_call.loc[df_call['NC_ID'] == 65].index, inplace=True)

In [113]:
df_call[df_call.NC_ID == 65]

Empty DataFrame
Columns: [SRNumber, CreatedDate, UpdatedDate, NC_ID, NCName, RequestType, RequestSource, Address, ZipCode, Latitude, Longitude, Location, PolicePrecinct]
Index: []

In [114]:
df_call[df_call.NC_ID == 117]

SRNumber             CreatedDate             UpdatedDate  NC_ID  \
387831   1-1342224881  04/30/2019 08:35:00 AM  05/07/2019 04:03:00 PM    117   
401888   1-1345024201  05/03/2019 08:46:00 AM  05/07/2019 04:02:00 PM    117   
403096   1-1345283454  05/03/2019 12:39:00 PM  05/04/2019 10:42:00 AM    117   
403617   1-1345434841  05/03/2019 02:49:00 PM  05/07/2019 04:06:00 PM    117   
404081   1-1345541671  05/03/2019 04:39:00 PM  05/07/2019 04:09:00 PM    117   
...               ...                     ...                     ...    ...   
1305371  1-1523011681  12/31/2019 09:09:00 AM  01/07/2020 12:52:00 PM    117   
1306718  1-1523302081  12/31/2019 01:01:00 PM  01/07/2020 07:39:00 PM    117   
1307105  1-1523425221  12/31/2019 02:36:00 PM  01/07/2020 06:00:00 PM    117   
1307937  1-1523562591  12/31/2019 07:11:00 PM  01/07/2020 05:04:00 PM    117   
1307939  1-1523562621  12/31/2019 07:12:00 PM  01/07/2020 08:03:00 PM    117   

                       NCName                 RequestType RequestSource  \
387831   PACIFIC PALISADES NC                 Bulky Items          Call   
401888   PACIFIC PALISADES NC                 Bulky Items          Call   
403096   PACIFIC PALISADES NC                       Other    Mobile App   
403617   PACIFIC PALISADES NC                 Bulky Items          Call   
404081   PACIFIC PALISADES NC                 Bulky Items          Call   
...                       ...                         ...           ...   
1305371  PACIFIC PALISADES NC            Electronic Waste          Call   
1306718  PACIFIC PALISADES NC      Illegal Dumping Pickup          Call   
1307105  PACIFIC PALISADES NC  Metal/Household Appliances  Self Service   
1307937  PACIFIC PALISADES NC                 Bulky Items          Call   
1307939  PACIFIC PALISADES NC  Metal/Household Appliances          Call   

                                 Address  ZipCode   Latitude   Longitude  \
387831           1149 N AMALFI DR, 90272  90272.0  34.050462 -118.507259   
401888          806 N HARTZELL ST, 90272  90272.0  34.043606 -118.519179   
403096    1260 N PIEDRA MORADA DR, 90272  90272.0  34.070012 -118.551481   
403617       16828 W CHARMEL LANE, 90272  90272.0  34.062258 -118.551652   
404081          674 N KINGMAN AVE, 90402  90402.0  34.036280 -118.508262   
...                                  ...      ...        ...         ...   
1305371         601 N FRONTERA DR, 90272  90272.0  34.039965 -118.522513   
1306718        15324 W DE PAUW ST, 90272  90272.0  34.039592 -118.526945   
1307105          207 N ENTRADA DR, 90402  90402.0  34.028928 -118.517226   
1307937  1435 N AVENIDA DE CORTEZ, 90272  90272.0  34.073532 -118.563304   
1307939  1435 N AVENIDA DE CORTEZ, 90272  90272.0  34.073532 -118.563304   

                                Location PolicePrecinct  
387831    (34.0504624342, -118.50725936)        WEST LA  
401888   (34.0436062925, -118.519179398)        WEST LA  
403096    (34.0700118216, -118.55148134)        WEST LA  
403617   (34.0622584212, -118.551652154)        WEST LA  
404081   (34.0362797109, -118.508261643)        WEST LA  
...                                  ...            ...  
1305371    (34.03996518, -118.522513312)        WEST LA  
1306718  (34.0395921447, -118.526945437)        WEST LA  
1307105  (34.0289282503, -118.517226485)        WEST LA  
1307937  (34.0735319431, -118.563304433)        WEST LA  
1307939  (34.0735319431, -118.563304433)        WEST LA  

[2818 rows x 13 columns]

In [115]:
#Doesn't exist.  Will delete.
df_call.drop(df_call.loc[df_call['NC_ID'] == 117].index, inplace=True)

In [116]:
df_call[df_call.NC_ID == 122]

SRNumber             CreatedDate             UpdatedDate  NC_ID  \
440338   1-1352862821  05/14/2019 05:06:00 PM  05/15/2019 08:20:00 AM    122   
457785   1-1356497201  05/20/2019 01:50:00 PM  05/23/2019 03:10:00 PM    122   
548048   1-1374696101  06/13/2019 04:57:00 PM  06/19/2019 07:22:00 AM    122   
548056   1-1374697321  06/13/2019 04:59:00 PM  06/19/2019 07:22:00 AM    122   
592248   1-1383618778  06/25/2019 04:54:00 PM  06/25/2019 04:55:00 PM    122   
621047   1-1389343676  07/02/2019 04:23:00 PM  07/02/2019 04:45:00 PM    122   
747364   1-1415482689  08/03/2019 02:57:00 AM  08/05/2019 11:05:00 AM    122   
747365   1-1415485850  08/03/2019 02:58:00 AM  08/05/2019 11:05:00 AM    122   
747366   1-1415483216  08/03/2019 02:58:00 AM  08/05/2019 11:05:00 AM    122   
773260   1-1420436894  08/09/2019 07:52:00 AM  08/15/2019 02:43:00 PM    122   
783500   1-1422226641  08/12/2019 01:48:00 PM  08/13/2019 01:33:00 PM    122   
790817   1-1423596861  08/13/2019 09:41:00 PM  08/19/2019 03:43:00 PM    122   
949399   1-1455558534  09/24/2019 09:01:00 AM  09/25/2019 02:31:00 PM    122   
958979   1-1457188142  09/26/2019 07:12:00 AM  09/27/2019 02:58:00 PM    122   
995717   1-1463760886  10/04/2019 06:07:00 PM  10/07/2019 04:41:00 PM    122   
1033640  1-1470267821  10/15/2019 06:57:00 AM  10/16/2019 02:52:00 PM    122   
1097675  1-1484743651  11/01/2019 02:46:00 PM  11/06/2019 03:32:00 PM    122   
1117552  1-1488218022  11/07/2019 08:57:00 AM  11/07/2019 02:27:00 PM    122   
1149648  1-1493739451  11/15/2019 10:32:32 AM  06/15/2020 11:30:35 AM    122   
1196873  1-1502063653  11/27/2019 05:04:00 PM  12/02/2019 05:19:00 PM    122   
1243509  1-1510701201  12/11/2019 11:07:00 AM  12/11/2019 11:10:00 AM    122   
1259856  1-1513859822  12/16/2019 09:52:00 AM  12/16/2019 05:40:00 PM    122   
1264251  1-1514689374  12/17/2019 07:02:00 AM  12/17/2019 03:27:00 PM    122   

                           NCName             RequestType       RequestSource  \
440338   HISTORIC FILIPINOTOWN NC        Electronic Waste                Call   
457785   HISTORIC FILIPINOTOWN NC        Graffiti Removal        Self Service   
548048   HISTORIC FILIPINOTOWN NC  Illegal Dumping Pickup                Call   
548056   HISTORIC FILIPINOTOWN NC             Bulky Items                Call   
592248   HISTORIC FILIPINOTOWN NC        Graffiti Removal  Driver Self Report   
621047   HISTORIC FILIPINOTOWN NC        Graffiti Removal  Driver Self Report   
747364   HISTORIC FILIPINOTOWN NC        Graffiti Removal          Mobile App   
747365   HISTORIC FILIPINOTOWN NC        Graffiti Removal          Mobile App   
747366   HISTORIC FILIPINOTOWN NC        Graffiti Removal          Mobile App   
773260   HISTORIC FILIPINOTOWN NC        Graffiti Removal          Mobile App   
783500   HISTORIC FILIPINOTOWN NC        Graffiti Removal                Call   
790817   HISTORIC FILIPINOTOWN NC        Graffiti Removal        Self Service   
949399   HISTORIC FILIPINOTOWN NC        Graffiti Removal          Mobile App   
958979   HISTORIC FILIPINOTOWN NC        Graffiti Removal          Mobile App   
995717   HISTORIC FILIPINOTOWN NC        Graffiti Removal          Mobile App   
1033640  HISTORIC FILIPINOTOWN NC        Graffiti Removal          Mobile App   
1097675  HISTORIC FILIPINOTOWN NC  Illegal Dumping Pickup                Call   
1117552  HISTORIC FILIPINOTOWN NC        Graffiti Removal          Mobile App   
1149648  HISTORIC FILIPINOTOWN NC     Homeless Encampment                Call   
1196873  HISTORIC FILIPINOTOWN NC        Graffiti Removal          Mobile App   
1243509  HISTORIC FILIPINOTOWN NC        Graffiti Removal  Driver Self Report   
1259856  HISTORIC FILIPINOTOWN NC        Graffiti Removal          Mobile App   
1264251  HISTORIC FILIPINOTOWN NC        Graffiti Removal          Mobile App   

                                     Address  ZipCode   Latitude   Longitude  \
440338   ROSELAKE AVE AT BEVERLY BLVD, 90057  90057.0  34.067490 -1

In [118]:
df_call[df_call.NC_ID == 122].count()

SRNumber          23
CreatedDate       23
UpdatedDate       23
NC_ID             23
NCName            23
RequestType       23
RequestSource     23
Address           23
ZipCode           23
Latitude          23
Longitude         23
Location          23
PolicePrecinct    23
dtype: int64

In [119]:
df_call.shape

(1292901, 13)

HISTORIC FILIPINOTOWN NC doesn't exist.  23 rows is still a tiny percentage of 1292901.  Will delete.

In [120]:
df_call[df_call.NC_ID == 123]

SRNumber             CreatedDate             UpdatedDate  NC_ID  \
401826   1-1344994011  05/03/2019 08:17:00 AM  05/09/2019 08:37:00 AM    123   
403472   1-1345392281  05/03/2019 02:19:00 PM  05/09/2019 08:45:00 AM    123   
404255   1-1345574911  05/03/2019 05:25:00 PM  05/23/2019 03:08:00 PM    123   
404551   1-1345618791  05/03/2019 08:36:00 PM  06/15/2019 08:48:00 AM    123   
404996   1-1345694448  05/04/2019 09:09:00 AM  05/09/2019 07:42:00 AM    123   
...               ...                     ...                     ...    ...   
1307684  1-1523527071  12/31/2019 04:24:00 PM  01/03/2020 03:42:00 PM    123   
1307827  1-1523542610  12/31/2019 05:10:00 PM  01/03/2020 03:40:00 PM    123   
1308050  1-1523581111  12/31/2019 09:38:00 PM  01/03/2020 03:43:00 PM    123   
1308053  1-1523583851  12/31/2019 09:40:00 PM  01/03/2020 03:44:00 PM    123   
1308056  1-1523584391  12/31/2019 09:44:00 PM  01/03/2020 04:30:00 PM    123   

                     NCName             RequestType RequestSource  \
401826   UNITED FOR VICTORY  Illegal Dumping Pickup          Call   
403472   UNITED FOR VICTORY             Bulky Items          Call   
404255   UNITED FOR VICTORY     Homeless Encampment  Self Service   
404551   UNITED FOR VICTORY  Illegal Dumping Pickup  Self Service   
404996   UNITED FOR VICTORY  Illegal Dumping Pickup    Mobile App   
...                     ...                     ...           ...   
1307684  UNITED FOR VICTORY             Bulky Items          Call   
1307827  UNITED FOR VICTORY             Bulky Items    Mobile App   
1308050  UNITED FOR VICTORY             Bulky Items  Self Service   
1308053  UNITED FOR VICTORY             Bulky Items  Self Service   
1308056  UNITED FOR VICTORY             Bulky Items  Self Service   

                             Address  ZipCode   Latitude   Longitude  \
401826   73RD ST AT HOOVER ST, 90044  90044.0  33.973633 -118.287125   
403472          834 W 73RD ST, 90044  90044.0  33.973460 -118.288392   
404255          839 W 73RD ST, 90044  90044.0  33.973739 -118.288523   
404551    6321 S ESTRELLA AVE, 90044  90044.0  33.981594 -118.285034   
404996     625 W FLORENCE AVE, 90044  90044.0  33.974736 -118.284364   
...                              ...      ...        ...         ...   
1307684         605 W 80TH ST, 90044  90044.0  33.966839 -118.284492   
1307827    6616 S FIGUEROA ST, 90003  90003.0  33.979239 -118.282514   
1308050      6509 S FLOWER ST, 90003  90003.0  33.980529 -118.281497   
1308053      6601 S FLOWER ST, 90003  90003.0  33.979782 -118.281497   
1308056      6409 S FLOWER ST, 90003  90003.0  33.981256 -118.281497   

                                Location PolicePrecinct  
401826   (33.9736328662, -118.287124532)    77TH STREET  
403472   (33.9734598783, -118.288391706)    77TH STREET  
404255   (33.9737389425, -118.288522634)    77TH STREET  
404551   (33.9815942245, -118.285033734)    77TH STREET  
404996   (33.9747359156, -118.284363859)    77TH STREET  
...                                  ...            ...  
1307684  (33.9668386585, -118.284492338)    77TH STREET  
1307827  (33.9792390471, -118.282514147)    77TH STREET  
1308050   (33.9805288556, -118.28149721)    77TH STREET  
1308053  (33.9797817182, -118.281496764)    77TH STREET  
1308056  (33.9812558632, -118.281496794)    77TH STREET  

[4075 rows x 13 columns]

In [122]:
#UNITED FOR VICTORY DOESN'T EXIST ANYMORE.  WILL DELETE.
df_call.drop(df_call.loc[df_call['NC_ID'] == 123].index, inplace=True)

In [124]:
NCDiff = np.setdiff1d(df_call.NC_ID,dfnc.NC_ID)
NCDiff

array([122])

df_call.NC_ID & dfnc.NC_ID are the same now

In [125]:
dfnc.NCName.value_counts()

CENTRAL ALAMEDA NC               1
GREATER VALLEY GLEN              1
NORTHRIDGE WEST NC               1
PANORAMA CITY NC                 1
NORTH HILLS EAST NC              1
                                ..
WESTSIDE NC                      1
ARTS DISTRICT LITTLE TOKYO NC    1
DOWNTOWN LOS ANGELES             1
GREATER WILSHIRE NC              1
NORTHRIDGE SOUTH NC              1
Name: NCName, Length: 99, dtype: int64

In [126]:
df_call.NCName.value_counts()

BOYLE HEIGHTS NC                           36689
SYLMAR NC                                  33699
VAN NUYS NC                                31200
EMPOWERMENT CONGRESS SOUTHEAST AREA NDC    30271
WILSHIRE CENTER - KOREATOWN NC             29340
                                           ...  
NORTH HILLS WEST                               2
GRANADA HILLS SOUTH                            1
LAKE BALBOA                                    1
ENCINO                                         1
RESEDA                                         1
Name: NCName, Length: 113, dtype: int64

Why are 113 unique values in df_call.NCName and 99 in dfnc?

In [127]:
df_call.NC_ID.value_counts()

50     36689
5      33699
20     31200
87     30271
24     29621
       ...  
46      3925
43      3119
127     2036
126     1062
122       23
Name: NC_ID, Length: 100, dtype: int64

In [128]:
NameDiff = np.setdiff1d(df_call.NCName,dfnc.NCName)
NameDiff

array(['CANOGA PARK', 'CHATSWORTH',
       'EMPOWERMENT CONGRESS CENTRAL AREA NDC',
       'EMPOWERMENT CONGRESS NORTH AREA NDC', 'ENCINO',
       'GRANADA HILLS SOUTH', 'GREATER ECHO PARK ELYSIAN NC',
       'GREATER VALLEY GLEN COUNCIL', 'HISTORIC CULTURAL NC',
       'HISTORIC FILIPINOTOWN NC', 'LAKE BALBOA', 'MAR VISTA CC',
       'MID-TOWN NORTH HOLLYWOOD NC', 'NC WESTCHESTER/PLAYA',
       'NORTH HILLS EAST', 'NORTH HILLS WEST',
       'NORTH HOLLYWOOD NORTHEAST NC', 'NORTHRIDGE EAST',
       'NORTHRIDGE WEST', 'PARK MESA HEIGHTS CC', 'PORTER RANCH',
       'RESEDA', 'UNITED NEIGHBORHOODS',
       'UNITED NEIGHBORHOODS OF THE HISTORIC ARLINGTON HEIGHTS, WEST ADAMS, AND JEFFERSON PARK COMMUNITY',
       'VOICES OF 90037', 'WEST HILLS', 'WEST LOS ANGELES NC', 'WINNETKA',
       'ZAPATA KING NC'], dtype=object)

In [132]:
dfnc[dfnc['NCName'] == 'CANOGA PARK']

Empty GeoDataFrame
Columns: [OBJECTID, NCName, WADDRESS, DWEBSITE, DEMAIL, DPHONE, NC_ID, CERTIFIED, TOOLTIP, NLA_URL, SERVICE_RE, geometry]
Index: []

In [136]:
dfnc[dfnc['NC_ID'] == 13]

OBJECTID          NCName                      WADDRESS  \
6         7  CANOGA PARK NC  http://www.canogaparknc.org/   

                    DWEBSITE              DEMAIL        DPHONE  NC_ID  \
6  http://empowerla.org/CPNC  CPNC@EmpowerLA.org  213-978-1551     13   

    CERTIFIED         TOOLTIP                                  NLA_URL  \
6  2002-06-18  CANOGA PARK NC  navigatela/reports/nc_reports.cfm?id=13   

                     SERVICE_RE  \
6  REGION 3 - SOUTH WEST VALLEY   

                                            geometry  
6  POLYGON ((-118.60465 34.23484, -118.60480 34.2...

Spelling differences but the numbers match

In [141]:
df_call.NC_ID.value_counts()

50     36689
5      33699
20     31200
87     30271
24     29621
       ...  
46      3925
43      3119
127     2036
126     1062
122       23
Name: NC_ID, Length: 100, dtype: int64

why 100 instead of 99?.  

In [144]:
df_call.shape

(1288826, 13)

In [145]:
type(df_call)

pandas.core.frame.DataFrame

In [146]:
dfnc.shape

(99, 12)

In [147]:
type(dfnc)

geopandas.geodataframe.GeoDataFrame

In [148]:
Geo_Call_Merge = dfnc.merge(df_call, on = 'NC_ID', how = 'outer', indicator = True)

In [149]:
type(Geo_Call_Merge)

geopandas.geodataframe.GeoDataFrame

In [150]:
Geo_Call_Merge.shape

(1288826, 25)

In [151]:
Geo_Call_Merge.head()

OBJECTID   NCName_x                  WADDRESS                  DWEBSITE  \
0       1.0  ARLETA NC  http://www.arletanc.org/  http://empowerla.org/ANC   
1       1.0  ARLETA NC  http://www.arletanc.org/  http://empowerla.org/ANC   
2       1.0  ARLETA NC  http://www.arletanc.org/  http://empowerla.org/ANC   
3       1.0  ARLETA NC  http://www.arletanc.org/  http://empowerla.org/ANC   
4       1.0  ARLETA NC  http://www.arletanc.org/  http://empowerla.org/ANC   

              DEMAIL        DPHONE  NC_ID   CERTIFIED    TOOLTIP  \
0  ANC@EmpowerLA.org  213-978-1551      6  2002-10-22  ARLETA NC   
1  ANC@EmpowerLA.org  213-978-1551      6  2002-10-22  ARLETA NC   
2  ANC@EmpowerLA.org  213-978-1551      6  2002-10-22  ARLETA NC   
3  ANC@EmpowerLA.org  213-978-1551      6  2002-10-22  ARLETA NC   
4  ANC@EmpowerLA.org  213-978-1551      6  2002-10-22  ARLETA NC   

                                  NLA_URL                    SERVICE_RE  \
0  navigatela/reports/nc_reports.cfm?id=6  REGION 1 - NORTH EAST VALLEY   
1  navigatela/reports/nc_reports.cfm?id=6  REGION 1 - NORTH EAST VALLEY   
2  navigatela/reports/nc_reports.cfm?id=6  REGION 1 - NORTH EAST VALLEY   
3  navigatela/reports/nc_reports.cfm?id=6  REGION 1 - NORTH EAST VALLEY   
4  navigatela/reports/nc_reports.cfm?id=6  REGION 1 - NORTH EAST VALLEY   

                                            geometry      SRNumber  \
0  POLYGON ((-118.44305 34.26328, -118.44303 34.2...  1-1262725561   
1  POLYGON ((-118.44305 34.26328, -118.44303 34.2...  1-1262780112   
2  POLYGON ((-118.44305 34.26328, -118.44303 34.2...  1-1262790302   
3  POLYGON ((-118.44305 34.26328, -118.44303 34.2...  1-1262791377   
4  POLYGON ((-118.44305 34.26328, -118.44303 34.2...  1-1262819459   

              CreatedDate             UpdatedDate   NCName_y  \
0  01/01/2019 10:30:00 AM  01/05/2019 11:16:00 AM  ARLETA NC   
1  01/01/2019 02:46:00 PM  01/02/2019 08:09:00 AM  ARLETA NC   
2  01/01/2019 03:31:00 PM  01/04/2019 12:21:00 PM  ARLETA NC   
3  01/01/2019 03:34:00 PM  01/08/2019 04:21:00 PM  ARLETA NC   
4  01/01/2019 05:29:00 PM  01/05/2019 11:54:00 AM  ARLETA NC   

              RequestType RequestSource                       Address  \
0             Bulky Items          Call  9489 N DORRINGTON AVE, 91331   
1        Graffiti Removal    Mobile App     14222 W WEIDNER ST, 91331   
2             Bulky Items    Mobile App    8940 N MERCEDES AVE, 91331   
3  Illegal Dumping Pickup    Mobile App   13154 W WENTWORTH ST, 91331   
4  Illegal Dumping Pickup    Mobile App  14066 W VAN NUYS BLVD, 91331   

   ZipCode   Latitude   Longitude                         Location  \
0  91331.0  34.242677 -118.444364  (34.2426768839, -118.444363615)   
1  91331.0  34.258201 -118.442768   (34.258200559, -118.442768159)   
2  91331.0  34.232185 -118.420323  (34.2321846692, -118.420322777)   
3  91331.0  34.231251 -118.419018  (34.2312506792, -118.419017925)   
4  91331.0  34.253178 -118.439457  (34.2531784699, -118.439456699)   

  PolicePrecinct _merge  
0        MISSION   both  
1        MISSION   both  
2        MISSION   both  
3        MISSION   both  
4        MISSION   both

In [152]:
Geo_Call_Merge._merge.value_counts()

both          1288803
right_only         23
left_only           0
Name: _merge, dtype: int64

In [154]:
Geo_Call_Merge[Geo_Call_Merge['_merge']=='right_only']

OBJECTID NCName_x WADDRESS DWEBSITE DEMAIL DPHONE  NC_ID CERTIFIED  \
1288803       NaN      NaN      NaN      NaN    NaN    NaN    122       NaN   
1288804       NaN      NaN      NaN      NaN    NaN    NaN    122       NaN   
1288805       NaN      NaN      NaN      NaN    NaN    NaN    122       NaN   
1288806       NaN      NaN      NaN      NaN    NaN    NaN    122       NaN   
1288807       NaN      NaN      NaN      NaN    NaN    NaN    122       NaN   
1288808       NaN      NaN      NaN      NaN    NaN    NaN    122       NaN   
1288809       NaN      NaN      NaN      NaN    NaN    NaN    122       NaN   
1288810       NaN      NaN      NaN      NaN    NaN    NaN    122       NaN   
1288811       NaN      NaN      NaN      NaN    NaN    NaN    122       NaN   
1288812       NaN      NaN      NaN      NaN    NaN    NaN    122       NaN   
1288813       NaN      NaN      NaN      NaN    NaN    NaN    122       NaN   
1288814       NaN      NaN      NaN      NaN    NaN    NaN    122       NaN   
1288815       NaN      NaN      NaN      NaN    NaN    NaN    122       NaN   
1288816       NaN      NaN      NaN      NaN    NaN    NaN    122       NaN   
1288817       NaN      NaN      NaN      NaN    NaN    NaN    122       NaN   
1288818       NaN      NaN      NaN      NaN    NaN    NaN    122       NaN   
1288819       NaN      NaN      NaN      NaN    NaN    NaN    122       NaN   
1288820       NaN      NaN      NaN      NaN    NaN    NaN    122       NaN   
1288821       NaN      NaN      NaN      NaN    NaN    NaN    122       NaN   
1288822       NaN      NaN      NaN      NaN    NaN    NaN    122       NaN   
1288823       NaN      NaN      NaN      NaN    NaN    NaN    122       NaN   
1288824       NaN      NaN      NaN      NaN    NaN    NaN    122       NaN   
1288825       NaN      NaN      NaN      NaN    NaN    NaN    122       NaN   

        TOOLTIP NLA_URL SERVICE_RE geometry      SRNumber  \
1288803     NaN     NaN        NaN     None  1-1352862821   
1288804     NaN     NaN        NaN     None  1-1356497201   
1288805     NaN     NaN        NaN     None  1-1374696101   
1288806     NaN     NaN        NaN     None  1-1374697321   
1288807     NaN     NaN        NaN     None  1-1383618778   
1288808     NaN     NaN        NaN     None  1-1389343676   
1288809     NaN     NaN        NaN     None  1-1415482689   
1288810     NaN     NaN        NaN     None  1-1415485850   
1288811     NaN     NaN        NaN     None  1-1415483216   
1288812     NaN     NaN        NaN     None  1-1420436894   
1288813     NaN     NaN        NaN     None  1-1422226641   
1288814     NaN     NaN        NaN     None  1-1423596861   
1288815     NaN     NaN        NaN     None  1-1455558534   
1288816     NaN     NaN        NaN     None  1-1457188142   
1288817     NaN     NaN        NaN     None  1-1463760886   
1288818     NaN     NaN        NaN     None  1-1470267821   
1288819     NaN     NaN        NaN     None  1-1484743651   
1288820     NaN     NaN        NaN     None  1-1488218022   
1288821     NaN     NaN        NaN     None  1-1493739451   
1288822     NaN     NaN        NaN     None  1-1502063653   
1288823     NaN     NaN        NaN     None  1-1510701201   
1288824     NaN     NaN        NaN     None  1-1513859822   
1288825     NaN     NaN        NaN     None  1-1514689374   

                    CreatedDate             UpdatedDate  \
1288803  05/14/2019 05:06:00 PM  05/15/2019 08:20:00 AM   
1288804  05/20/2019 01:50:00 PM  05/23/2019 03:10:00 PM   
1288805  06/13/2019 04:57:00 PM  06/19/2019 07:22:00 AM   
1288806  06/13/2019 04:59:00 PM  06/19/2019 07:22:00 AM   
1288807  06/25/2019 04:54:00 PM  06/25/2019 04:55:00 PM   
1288808  07/02/2019 04:23:00 PM  07/02/2019 04:45:00 PM   
1288809  08/03/2019 02:57:00 AM  08/05/2019 11:05:00 AM   
1288810  08/03/2019 02:58:00 AM  08/05/2019 11:05:00 AM   
1288811  08/03/2019 02:58:00 AM  08/05/2019 11:05:00 AM   
1288812  08/09/2019 07:52:00 AM  08/15/2019 02:43:00 PM  

These are all in a NC that doesn't exist anymore.  Will delete.

In [155]:
df_call.drop(df_call.loc[df_call['NC_ID'] == 2].index, inplace=True)
Geo_Call_Merge.drop(Geo_Call_Merge.loc[Geo_Call_Merge['_merge']=='right_only'].index, inplace=True)

In [158]:
Geo_Call_Merge['_merge'].value_counts()

both          1288803
left_only           0
right_only          0
Name: _merge, dtype: int64

In [164]:
#now that the 'left_only' rows are gone, delete the merge indicator  (Error below because accidently ran twice)
Geo_Call_Merge.drop(columns='_merge', inplace=True)

KeyError: "['_merge'] not found in axis"

In [163]:
Geo_Call_Merge.head(2)

OBJECTID   NCName_x                  WADDRESS                  DWEBSITE  \
0       1.0  ARLETA NC  http://www.arletanc.org/  http://empowerla.org/ANC   
1       1.0  ARLETA NC  http://www.arletanc.org/  http://empowerla.org/ANC   

              DEMAIL        DPHONE  NC_ID   CERTIFIED    TOOLTIP  \
0  ANC@EmpowerLA.org  213-978-1551      6  2002-10-22  ARLETA NC   
1  ANC@EmpowerLA.org  213-978-1551      6  2002-10-22  ARLETA NC   

                                  NLA_URL                    SERVICE_RE  \
0  navigatela/reports/nc_reports.cfm?id=6  REGION 1 - NORTH EAST VALLEY   
1  navigatela/reports/nc_reports.cfm?id=6  REGION 1 - NORTH EAST VALLEY   

                                            geometry      SRNumber  \
0  POLYGON ((-118.44305 34.26328, -118.44303 34.2...  1-1262725561   
1  POLYGON ((-118.44305 34.26328, -118.44303 34.2...  1-1262780112   

              CreatedDate             UpdatedDate   NCName_y  \
0  01/01/2019 10:30:00 AM  01/05/2019 11:16:00 AM  ARLETA NC   
1  01/01/2019 02:46:00 PM  01/02/2019 08:09:00 AM  ARLETA NC   

        RequestType RequestSource                       Address  ZipCode  \
0       Bulky Items          Call  9489 N DORRINGTON AVE, 91331  91331.0   
1  Graffiti Removal    Mobile App     14222 W WEIDNER ST, 91331  91331.0   

    Latitude   Longitude                         Location PolicePrecinct  
0  34.242677 -118.444364  (34.2426768839, -118.444363615)        MISSION  
1  34.258201 -118.442768   (34.258200559, -118.442768159)        MISSION

In [165]:
#Delete NCName_x and delete NCName_y
Geo_Call_Merge.rename(columns = {"NCName_x": "NCName"}, inplace=True)

In [168]:
Geo_Call_Merge.drop(columns='NCName_y', inplace=True)

In [170]:
Geo_Call_Merge.head(2)

OBJECTID     NCName                  WADDRESS                  DWEBSITE  \
0       1.0  ARLETA NC  http://www.arletanc.org/  http://empowerla.org/ANC   
1       1.0  ARLETA NC  http://www.arletanc.org/  http://empowerla.org/ANC   

              DEMAIL        DPHONE  NC_ID   CERTIFIED    TOOLTIP  \
0  ANC@EmpowerLA.org  213-978-1551      6  2002-10-22  ARLETA NC   
1  ANC@EmpowerLA.org  213-978-1551      6  2002-10-22  ARLETA NC   

                                  NLA_URL                    SERVICE_RE  \
0  navigatela/reports/nc_reports.cfm?id=6  REGION 1 - NORTH EAST VALLEY   
1  navigatela/reports/nc_reports.cfm?id=6  REGION 1 - NORTH EAST VALLEY   

                                            geometry      SRNumber  \
0  POLYGON ((-118.44305 34.26328, -118.44303 34.2...  1-1262725561   
1  POLYGON ((-118.44305 34.26328, -118.44303 34.2...  1-1262780112   

              CreatedDate             UpdatedDate       RequestType  \
0  01/01/2019 10:30:00 AM  01/05/2019 11:16:00 AM       Bulky Items   
1  01/01/2019 02:46:00 PM  01/02/2019 08:09:00 AM  Graffiti Removal   

  RequestSource                       Address  ZipCode   Latitude   Longitude  \
0          Call  9489 N DORRINGTON AVE, 91331  91331.0  34.242677 -118.444364   
1    Mobile App     14222 W WEIDNER ST, 91331  91331.0  34.258201 -118.442768   

                          Location PolicePrecinct  
0  (34.2426768839, -118.444363615)        MISSION  
1   (34.258200559, -118.442768159)        MISSION

The merge worked correctly.  Will delete repetitive rows and info that can easily be found on the LA Call website and save the dataset and start a new Jupyter Notebook to merge the dataset with the crime dataset.

In [171]:
Geo_Call_Merge.drop(columns=['WADDRESS', 'DWEBSITE', 'DEMAIL', 'DPHONE'], inplace=True)

In [172]:
Geo_Call_Merge.head(2)

OBJECTID     NCName  NC_ID   CERTIFIED    TOOLTIP  \
0       1.0  ARLETA NC      6  2002-10-22  ARLETA NC   
1       1.0  ARLETA NC      6  2002-10-22  ARLETA NC   

                                  NLA_URL                    SERVICE_RE  \
0  navigatela/reports/nc_reports.cfm?id=6  REGION 1 - NORTH EAST VALLEY   
1  navigatela/reports/nc_reports.cfm?id=6  REGION 1 - NORTH EAST VALLEY   

                                            geometry      SRNumber  \
0  POLYGON ((-118.44305 34.26328, -118.44303 34.2...  1-1262725561   
1  POLYGON ((-118.44305 34.26328, -118.44303 34.2...  1-1262780112   

              CreatedDate             UpdatedDate       RequestType  \
0  01/01/2019 10:30:00 AM  01/05/2019 11:16:00 AM       Bulky Items   
1  01/01/2019 02:46:00 PM  01/02/2019 08:09:00 AM  Graffiti Removal   

  RequestSource                       Address  ZipCode   Latitude   Longitude  \
0          Call  9489 N DORRINGTON AVE, 91331  91331.0  34.242677 -118.444364   
1    Mobile App     14222 W WEIDNER ST, 91331  91331.0  34.258201 -118.442768   

                          Location PolicePrecinct  
0  (34.2426768839, -118.444363615)        MISSION  
1   (34.258200559, -118.442768159)        MISSION

In [173]:
Geo_Call_Merge.drop(columns='NLA_URL', inplace=True)

In [174]:
Geo_Call_Merge.head(2)

OBJECTID     NCName  NC_ID   CERTIFIED    TOOLTIP  \
0       1.0  ARLETA NC      6  2002-10-22  ARLETA NC   
1       1.0  ARLETA NC      6  2002-10-22  ARLETA NC   

                     SERVICE_RE  \
0  REGION 1 - NORTH EAST VALLEY   
1  REGION 1 - NORTH EAST VALLEY   

                                            geometry      SRNumber  \
0  POLYGON ((-118.44305 34.26328, -118.44303 34.2...  1-1262725561   
1  POLYGON ((-118.44305 34.26328, -118.44303 34.2...  1-1262780112   

              CreatedDate             UpdatedDate       RequestType  \
0  01/01/2019 10:30:00 AM  01/05/2019 11:16:00 AM       Bulky Items   
1  01/01/2019 02:46:00 PM  01/02/2019 08:09:00 AM  Graffiti Removal   

  RequestSource                       Address  ZipCode   Latitude   Longitude  \
0          Call  9489 N DORRINGTON AVE, 91331  91331.0  34.242677 -118.444364   
1    Mobile App     14222 W WEIDNER ST, 91331  91331.0  34.258201 -118.442768   

                          Location PolicePrecinct  
0  (34.2426768839, -118.444363615)        MISSION  
1   (34.258200559, -118.442768159)        MISSION

In [175]:
Geo_Call_Merge.isnull().sum()

OBJECTID          0
NCName            0
NC_ID             0
CERTIFIED         0
TOOLTIP           0
SERVICE_RE        0
geometry          0
SRNumber          0
CreatedDate       0
UpdatedDate       0
RequestType       0
RequestSource     0
Address           0
ZipCode           0
Latitude          0
Longitude         0
Location          0
PolicePrecinct    0
dtype: int64

In [178]:
Geo_Call_Merge[Geo_Call_Merge.duplicated()]

Empty GeoDataFrame
Columns: [OBJECTID, NCName, NC_ID, CERTIFIED, TOOLTIP, SERVICE_RE, geometry, SRNumber, CreatedDate, UpdatedDate, RequestType, RequestSource, Address, ZipCode, Latitude, Longitude, Location, PolicePrecinct]
Index: []

In [179]:
Geo_Call_Merge.CreatedDate.max()

'12/31/2019 12:59:00 PM'

In [180]:
Geo_Call_Merge.CreatedDate.min()

'01/01/2019 01:01:00 PM'

In [181]:
Geo_Call_Merge.UpdatedDate.max()

'12/31/2019 12:59:00 PM'

In [182]:
Geo_Call_Merge.UpdatedDate.min()

'01/01/2019 01:03:00 PM'

Save Merge file

get rid of errors

In [185]:
Geo_Call_Merge.head(2)

OBJECTID     NCName  NC_ID   CERTIFIED    TOOLTIP  \
0       1.0  ARLETA NC      6  2002-10-22  ARLETA NC   
1       1.0  ARLETA NC      6  2002-10-22  ARLETA NC   

                     SERVICE_RE  \
0  REGION 1 - NORTH EAST VALLEY   
1  REGION 1 - NORTH EAST VALLEY   

                                            geometry      SRNumber  \
0  POLYGON ((-118.44305 34.26328, -118.44303 34.2...  1-1262725561   
1  POLYGON ((-118.44305 34.26328, -118.44303 34.2...  1-1262780112   

              CreatedDate             UpdatedDate       RequestType  \
0  01/01/2019 10:30:00 AM  01/05/2019 11:16:00 AM       Bulky Items   
1  01/01/2019 02:46:00 PM  01/02/2019 08:09:00 AM  Graffiti Removal   

  RequestSource                       Address  ZipCode   Latitude   Longitude  \
0          Call  9489 N DORRINGTON AVE, 91331  91331.0  34.242677 -118.444364   
1    Mobile App     14222 W WEIDNER ST, 91331  91331.0  34.258201 -118.442768   

                          Location PolicePrecinct  
0  (34.2426768839, -118.444363615)        MISSION  
1   (34.258200559, -118.442768159)        MISSION

In [187]:
Geo_Call_Merge.rename(columns = {'CreatedDate' : 'CreatDt', 'UpdatedDate': 'UpdateDt', 'RequestType': 'ReqTpe' , 'RequestSource': 'ReqSrc', 'Latitude': 'Lat', 'Longitude': 'Lon', 'PolicePrecinct': 'Area'}, inplace = True)

In [188]:
Geo_Call_Merge.head(2)

OBJECTID     NCName  NC_ID   CERTIFIED    TOOLTIP  \
0       1.0  ARLETA NC      6  2002-10-22  ARLETA NC   
1       1.0  ARLETA NC      6  2002-10-22  ARLETA NC   

                     SERVICE_RE  \
0  REGION 1 - NORTH EAST VALLEY   
1  REGION 1 - NORTH EAST VALLEY   

                                            geometry      SRNumber  \
0  POLYGON ((-118.44305 34.26328, -118.44303 34.2...  1-1262725561   
1  POLYGON ((-118.44305 34.26328, -118.44303 34.2...  1-1262780112   

                  CreatDt                UpdateDt            ReqTpe  \
0  01/01/2019 10:30:00 AM  01/05/2019 11:16:00 AM       Bulky Items   
1  01/01/2019 02:46:00 PM  01/02/2019 08:09:00 AM  Graffiti Removal   

       ReqSrc                       Address  ZipCode        Lat         Lon  \
0        Call  9489 N DORRINGTON AVE, 91331  91331.0  34.242677 -118.444364   
1  Mobile App     14222 W WEIDNER ST, 91331  91331.0  34.258201 -118.442768   

                          Location     Area  
0  (34.2426768839, -118.444363615)  MISSION  
1   (34.258200559, -118.442768159)  MISSION

In [199]:

outpickle = r'C:\Documents\projects\HackLA\311\data\WorkedonData\Geo\call_NC_merge_2019.pickle'

In [200]:

Geo_Call_Merge.to_pickle(outpickle)

In [201]:
geo_test = pd.read_pickle(outpickle)
df2 = pd.read_pickle('my_df.pickle')

df2

In [202]:
geo_test.head()

OBJECTID     NCName  NC_ID   CERTIFIED    TOOLTIP  \
0       1.0  ARLETA NC      6  2002-10-22  ARLETA NC   
1       1.0  ARLETA NC      6  2002-10-22  ARLETA NC   
2       1.0  ARLETA NC      6  2002-10-22  ARLETA NC   
3       1.0  ARLETA NC      6  2002-10-22  ARLETA NC   
4       1.0  ARLETA NC      6  2002-10-22  ARLETA NC   

                     SERVICE_RE  \
0  REGION 1 - NORTH EAST VALLEY   
1  REGION 1 - NORTH EAST VALLEY   
2  REGION 1 - NORTH EAST VALLEY   
3  REGION 1 - NORTH EAST VALLEY   
4  REGION 1 - NORTH EAST VALLEY   

                                            geometry      SRNumber  \
0  POLYGON ((-118.44305 34.26328, -118.44303 34.2...  1-1262725561   
1  POLYGON ((-118.44305 34.26328, -118.44303 34.2...  1-1262780112   
2  POLYGON ((-118.44305 34.26328, -118.44303 34.2...  1-1262790302   
3  POLYGON ((-118.44305 34.26328, -118.44303 34.2...  1-1262791377   
4  POLYGON ((-118.44305 34.26328, -118.44303 34.2...  1-1262819459   

                  CreatDt                UpdateDt                  ReqTpe  \
0  01/01/2019 10:30:00 AM  01/05/2019 11:16:00 AM             Bulky Items   
1  01/01/2019 02:46:00 PM  01/02/2019 08:09:00 AM        Graffiti Removal   
2  01/01/2019 03:31:00 PM  01/04/2019 12:21:00 PM             Bulky Items   
3  01/01/2019 03:34:00 PM  01/08/2019 04:21:00 PM  Illegal Dumping Pickup   
4  01/01/2019 05:29:00 PM  01/05/2019 11:54:00 AM  Illegal Dumping Pickup   

       ReqSrc                       Address  ZipCode        Lat         Lon  \
0        Call  9489 N DORRINGTON AVE, 91331  91331.0  34.242677 -118.444364   
1  Mobile App     14222 W WEIDNER ST, 91331  91331.0  34.258201 -118.442768   
2  Mobile App    8940 N MERCEDES AVE, 91331  91331.0  34.232185 -118.420323   
3  Mobile App   13154 W WENTWORTH ST, 91331  91331.0  34.231251 -118.419018   
4  Mobile App  14066 W VAN NUYS BLVD, 91331  91331.0  34.253178 -118.439457   

                          Location     Area  
0  (34.2426768839, -118.444363615)  MISSION  
1   (34.258200559, -118.442768159)  MISSION  
2  (34.2321846692, -118.420322777)  MISSION  
3  (34.2312506792, -118.419017925)  MISSION  
4  (34.2531784699, -118.439456699)  MISSION

Pickle works!

Will save this notebook and use a fresh one to merge the call & geometry file with the crime dataset